pom.xml

```diff
@@ -119,18 +125,201 @@
+            <!-- Use Unix and Mac style line endings -->
+            <lineEnding>LF</lineEnding>
+            <encoding>UTF-8</encoding>
+            <skipJsFormatting>true</skipJsFormatting>
+            <skipCssFormatting>true</skipCssFormatting>
+            <skipHtmlFormatting>true</skipHtmlFormatting>
+            <skipJsonFormatting>true</skipJsonFormatting>
+            <skipXmlFormatting>true</skipXmlFormatting>
+            <skipJavaFormatting>false</skipJavaFormatting>
+          </configuration>
         </plugin>
         <plugin>
           <groupId>com.github.ekryd.sortpom</groupId>
           <artifactId>sortpom-maven-plugin</artifactId>
           <version>2.15.0</version>
+          <executions>
+            <execution>
+              <phase>verify</phase>
+              <goals>
+                <goal>${pom.fmt.action}</goal>
+              </goals>
+            </execution>
+          </executions>
+          <configuration>
+            <skip>${pom.fmt.skip}</skip>
+            <pomFile>pom.xml</pomFile>
+            <keepBlankLines>true</keepBlankLines>
+            <spaceBeforeCloseEmptyElement>false</spaceBeforeCloseEmptyElement>
+            <createBackupFile>false</createBackupFile>
+            <verifyFail>stop</verifyFail>
+            <verifyFailOn>strict</verifyFailOn>
+          </configuration>
         </plugin>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.14.0</version>
+          <configuration>
+            <annotationProcessorPaths>
+              <path>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-configuration-processor</artifactId>
+              </path>
+              <path>
+                <groupId>org.projectlombok</groupId>
+                <artifactId>lombok</artifactId>
+                <version>${lombok.version}</version>
+              </path>
+            </annotationProcessorPaths>
+          </configuration>
+        </plugin>
+        <plugin>
+          <groupId>org.jacoco</groupId>
+          <artifactId>jacoco-maven-plugin</artifactId>
+          <version>0.8.12</version>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <configuration>
+            <argLine>-javaagent:${settings.localRepository}/org/mockito/mockito-core/${mockito.version}/mockito-core-${mockito.version}.jar
+            -Xshare:off</argLine>
+          </configuration>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-failsafe-plugin</artifactId>
+          <configuration>
+            <argLine>-javaagent:${settings.localRepository}/org/mockito/mockito-core/${mockito.version}/mockito-core-${mockito.version}.jar
+            -Xshare:off</argLine>
```

Comment:
What does the mockito agent provide to the test environment ? (sorry for the dumb question, but this is the first time I see it configured)

Reply:
Good question @pmauduit!

This is the recommended Mockito setup for newer Java versions (17+) and Spring Boot 3.x. It enables mocking of final classes and methods, for testing with Spring Boot 3, which has increased its usage of final classes for performance.

`-Xshare:off` prevents conflicts when Mockito manipulates bytecode